pkgcore/ebuild/processor.py

```diff
@@ -347,17 +347,18 @@ def __init__(self, userpriv, sandbox, fakeroot, save_file):
         # force to a neutral dir so that sandbox/fakeroot won't explode if
         # ran from a nonexistent dir
         spawn_opts["cwd"] = e_const.EAPI_BIN_PATH
-        # little trick. we force the pipes to be high up fd wise so
-        # nobody stupidly hits 'em.
-        max_fd = min(pkgcore.spawn.max_fd_limit, 1024)
+
+        # Use consecutive FDs because deriving from RLIMIT_NOFILE yields unexpected results on OSX.
+        read_fd = 3
```

Comment:
Just to be clear, I don't see any relevant code in Portage. The only occurence of NOFILE is used to scan for open fds, and this in turn seems to be used only to close them.